stdlib/public/core/Collection.swift

```diff
@@ -24,7 +24,8 @@ public protocol Indexable {
   ///
   /// Valid indices consist of the position of every element and a
   /// "past the end" position that's not valid for use as a subscript.
-  associatedtype Index : ForwardIndex
+// TODO: swift-3-indexing-model - Index only needs to be comparable or must be comparable..?
```

Comment:
I am not sure I understand the question.  Our thinking was that it can always be comparable, so we are adding a requirement.  We would be interested to know if you can think of a collection where it wouldn't be possible to make index comparable in O(1).